### PR TITLE
[i18n-ar] Translated file: `docs/source/ar/tasks/question_answering.md` into Arabic

### PR DESCRIPTION
**Pull request by AhmedAlmaghz**
## What does this PR do?
Translated the `docs/source/ar/tasks/question_answering.md` file of the documentation to `Arabic العربية`. I use transformers for my side-project and wanted to contribute. As a beginner, this seemed appropriate.

### Some questions:

- I can't access the pr documentation endpoint for `Arabic العربية`. Is there a missing step?
- I left all the other files alone after copying the English docs. Is this acceptable for future updates or should I add contents to the `_toctree.yml` only when translation is complete? Should I delete all other files for a cleaner review?
- In `Arabic`, people use both `Transformer` and `المحولات` to describe the model. However, as this is more of a product of HuggingFace I opted not to translate it. May you please let me know which you would prefer?

Thank you in advance for your review.

Part of #32435

### Before submitting

- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [ ] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#start-contributing-pull-requests), Pull Request section?
- [ ] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/) ?Please add a link to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

### Who can review?
@stevhliu , @MKhalusova , @abodacs, may you please review this PR?

### CODE REVIEW

File: docs/source/ar/tasks/question_answering.md [hunk 1]
```diff
@@ -0,0 +1,432 @@
...
+
+```py
+>>> from huggingface_hub import notebook_login
+
+>>> notebook_login()
+```
+
+## تحميل مجموعة بيانات SQuAD
+
+ابدأ بتحميل مجموعة فرعية أصغر من مجموعة بيانات SQuAD من مكتبة 🤗 Datasets. سيعطيك هذا فرصة للتجربة والتأكد من أن كل شيء يعمل قبل قضاء المزيد من الوقت في التدريب على مجموعة البيانات الكاملة.
```
Comment (abodacs) [inline docs/source/ar/tasks/question_answering.md:55]:
```suggestion
ابدأ بتحميل جزء أصغر من مجموعة بيانات SQuAD من مكتبة 🤗 Datasets. سيتيح لك ذلك فرصة للتجربة والتحقق من عمل كل شيء بشكل صحيح قبل قضاء المزيد من الوقت في التدريب على مجموعة البيانات الكاملة.
```

File: docs/source/ar/tasks/question_answering.md [hunk 2]
```diff
@@ -0,0 +1,432 @@
...
+
+- `answers`: موقع بداية الرمز المميز للإجابة ونص الإجابة.
+- `context`: معلومات أساسية يحتاج النموذج إلى استخراج الإجابة منها.
+- `question`: السؤال الذي يجب على النموذج الإجابة عليه.
+
+## المعالجة المسبقة (Preprocess)
+
+<Youtube id="qgaM0weJHpA"/>
+
+الخطوة التالية هي تحميل محلل DistilBERT لمعالجة حقلي `question` و `context`:
```
Comment (abodacs) [inline docs/source/ar/tasks/question_answering.md:91]:
```suggestion
الخطوة التالية هي تحميل المحلل اللغوى DistilBERT لمعالجة حقلي `question` و `context`:
```

File: docs/source/ar/tasks/question_answering.md [hunk 3]
```diff
@@ -0,0 +1,432 @@
...
+
+هناك بعض خطوات المعالجة المسبقة الخاصة بمهام الإجابة على الأسئلة التي يجب أن تكون على دراية بها:
+
+1. قد تحتوي بعض الأمثلة في مجموعة البيانات على `context` طويل جدًا يتجاوز الحد الأقصى لطول إدخال النموذج. للتعامل مع التسلسلات الأطول، قم بقص `context` فقط عن طريق تعيين `truncation="only_second"`.
+2. بعد ذلك، قم بتعيين مواضع البداية والنهاية للإجابة إلى `context` الأصلي عن طريق تعيين
+   `return_offset_mapping=True`.
+3. مع وجود التعيين في متناول اليد، يمكنك الآن العثور على الرموز المميزة لبداية ونهاية الإجابة. استخدم طريقة [`~tokenizers.Encoding.sequence_ids`]
+   للعثور على أي جزء من الإزاحة يتوافق مع `question` وأي جزء يتوافق مع `context`.
+
+فيما يلي كيفية إنشاء دالة لقص وتعيين رموز البداية والنهاية لـ `answer` إلى `context`:
```
Comment (abodacs) [inline docs/source/ar/tasks/question_answering.md:107]:
```suggestion
1. قد تحتوي بعض الأمثلة في مجموعة البيانات على `context` طويلًا يتجاوز الحد الأقصى لطول مدخل النموذج. للتعامل مع النصوص الأطول، يتم اقتطاع `context` فقط عن طريق تعيين `truncation="only_second"`.
2. بعد ذلك، يتم تحديد مواضع بداية ونهاية الإجابة في `context` الأصلي عن طريق تعيين
   `return_offset_mapping=True`.
3. باستخدام التعيين، يمكن الآن تحديد رموز بداية ونهاية الإجابة. استخدم طريقة [`~tokenizers.Encoding.sequence_ids`]
   لتحديد أجزاء الإزاحة التي تتوافق مع `question` و `context`.

فيما يلي كيفية إنشاء دالة لقص وتعيين رموز البداية والنهاية لـ `answer` إلى `context`:
```

File: docs/source/ar/tasks/question_answering.md [hunk 4]
```diff
@@ -0,0 +1,432 @@
...
+...             while idx >= context_start and offset[idx][1] >= end_char:
+...                 idx -= 1
+...             end_positions.append(idx + 1)
+
+...     inputs["start_positions"] = start_positions
+...     inputs["end_positions"] = end_positions
+...     return inputs
+```
+
+لتطبيق دالة المعالجة المسبقة على مجموعة البيانات بأكملها، استخدم دالة [`~datasets.Dataset.map`] من 🤗 Datasets. يمكنك تسريع دالة `map` عن طريق تعيين `batched=True` لمعالجة عناصر متعددة من مجموعة البيانات في وقت واحد. قم بإزالة أي أعمدة لا تحتاجها:
```
Comment (abodacs) [inline docs/source/ar/tasks/question_answering.md:162]:
```suggestion
لتطبيق المعالجة المسبقة على كامل مجموعة البيانات، استخدم [`~datasets.Dataset.map`] من مكتبة 🤗 Datasets. يمكنك تسريع دالة `map` عن طريق تعيين `batched=True` لمعالجة عناصر متعددة من مجموعة البيانات دفعة واحدة. قم بإزالة أي أعمدة لا تحتاجها:
```

File: docs/source/ar/tasks/question_answering.md [hunk 5]
```diff
@@ -0,0 +1,432 @@
...
+
+```py
+>>> tokenized_squad = squad.map(preprocess_function, batched=True, remove_columns=squad["train"].column_names)
+```
+
+الآن قم بإنشاء دفعة من الأمثلة باستخدام [`DefaultDataCollator`]. على عكس أدوات تجميع البيانات الأخرى في 🤗 Transformers، لا يطبق [`DefaultDataCollator`] أي معالجة مسبقة إضافية مثل الحشو.
```
Comment (abodacs) [inline docs/source/ar/tasks/question_answering.md:168]:
```suggestion
الآن قم بإنشاء دفعة من الأمثلة باستخدام [`DefaultDataCollator`]. بخلاف مجمّعات البيانات الأخرى في 🤗 Transformers، لا يطبق [`DefaultDataCollator`] أي معالجة مسبقة إضافية مثل الحشو.
```

File: docs/source/ar/tasks/question_answering.md [hunk 6]
```diff
@@ -0,0 +1,432 @@
...
+>>> from transformers import AutoModelForQuestionAnswering, TrainingArguments, Trainer
+
+>>> model = AutoModelForQuestionAnswering.from_pretrained("distilbert/distilbert-base-uncased")
+```
+
+في هذه المرحلة، تبقى ثلاث خطوات فقط:
+
+1. حدد المعاملات الفائقة للتدريب في [`TrainingArguments`]. المعامل الوحيد المطلوب هو `output_dir` الذي يحدد مكان حفظ نموذجك. ستدفع هذا النموذج إلى Hub عن طريق تعيين `push_to_hub=True` (يجب عليك تسجيل الدخول إلى Hugging Face لتحميل نموذجك).
+2. مرر معاملات التدريب إلى [`Trainer`] جنبًا إلى جنب مع النموذج، ومجموعة البيانات، والمعالج المعجمي، وجامع البيانات.
+3. اتصل بـ [`~Trainer.train`] لضبط نموذجك.
```
Comment (abodacs) [inline docs/source/ar/tasks/question_answering.md:212]:
```suggestion
1. حدد المعاملات الفائقة للتدريب في [`TrainingArguments`]. المعامل الوحيد المطلوب هو `output_dir` الذي يحدد مكان حفظ نموذجك. ستدفع هذا النموذج إلى Hub عن طريق تعيين `push_to_hub=True` (يجب عليك تسجيل الدخول إلى Hugging Face لتحميل نموذجك).
2. مرر معاملات التدريب إلى [`Trainer`] جنبًا إلى جنب مع النموذج، ومجموعة البيانات، والمُحلّل النصي، ومُجمّع البيانات.
3. استدعِ ـ [`~Trainer.train`] لضبط النموذج.
```

File: docs/source/ar/tasks/question_answering.md [hunk 7]
```diff
@@ -0,0 +1,432 @@
...
+...     train_dataset=tokenized_squad["train"],
+...     eval_dataset=tokenized_squad["test"],
+...     processing_class=tokenizer,
+...     data_collator=data_collator,
+... )
+
+>>> trainer.train()
+```
+
+بمجرد اكتمال التدريب، شارك نموذجك في Hub باستخدام طريقة [`~transformers.Trainer.push_to_hub`] حتى يتمكن الجميع من استخدام نموذجك:
```
Comment (abodacs) [inline docs/source/ar/tasks/question_answering.md:238]:
```suggestion
بمجرد اكتمال التدريب، شارك نموذجك في Hub باستخدام الدالة [`~transformers.Trainer.push_to_hub`] حتى يتمكن الجميع من استخدام نموذجك:
```

File: docs/source/ar/tasks/question_answering.md [hunk 8]
```diff
@@ -0,0 +1,432 @@
...
+أو [دفتر ملاحظات TensorFlow](https://colab.research.google.com/github/huggingface/notebooks/blob/main/examples/question_answering-tf.ipynb).
+
+</Tip>
+
+## التقييم (Evaluate)
+
+يتطلب التقييم للإجابة على الأسئلة قدرًا كبيرًا من المعالجة اللاحقة. لتجنب استغراق الكثير من وقتك، يتخطى هذا الدليل خطوة التقييم. لا يزال [`Trainer`] يحسب خسارة التقييم أثناء التدريب، لذا فأنت لست في الظلام تمامًا بشأن أداء نموذجك.
+
+إذا كان لديك المزيد من الوقت وتهتم بكيفية تقييم نموذجك للإجابة على الأسئلة، فألق نظرة على فصل [الإجابة على الأسئلة](https://huggingface.co/course/chapter7/7?fw=pt#post-processing) من دورة 🤗 Hugging Face!
+
```
Comment (abodacs) [inline docs/source/ar/tasks/question_answering.md:333]:
```suggestion
يتطلب التقييم للإجابة على الأسئلة قدرًا كبيرًا من المعالجة اللاحقة. لتوفير وقتك، يتخطى هذا الدليل خطوة التقييم. لا يزال [`Trainer`] يحسب خسارة التقييم أثناء التدريب، مما يعني أنك لست تجهل تمامًا أداء نموذجك.

إذا كان لديك المزيد من الوقت وتهتم بكيفية تقييم نموذجك للإجابة على الأسئلة، فألق نظرة على فصل [الإجابة على الأسئلة](https://huggingface.co/course/chapter7/7?fw=pt#post-processing) من دورة 🤗 Hugging Face!

```

File: docs/source/ar/tasks/question_answering.md [hunk 9]
```diff
@@ -0,0 +1,432 @@
...
+## الاستدلال (Inference)
+
+رائع، الآن بعد أن قمت بضبط نموذج، يمكنك استخدامه للاستدلال!
+
+ابتكر سؤالًا وسياقًا ترغب في أن يتنبأ به النموذج:
```
Comment (abodacs) [inline docs/source/ar/tasks/question_answering.md:338]:
```suggestion
حدد سؤالًا وسياقًا ليقوم النموذج بالتنبؤ بالإجابة عليه:
```

File: docs/source/ar/tasks/question_answering.md [hunk 10]
```diff
@@ -0,0 +1,432 @@
...
+
+```py
+>>> question = "How many programming languages does BLOOM support?"
+>>> context = "BLOOM has 176 billion parameters and can generate text in 46 languages natural languages and 13 programming languages."
+```
+
+أبسط طريقة لتجربة نموذجك المضبوط للاستدلال هي استخدامه في [`pipeline`]. قم بإنشاء مثيل لـ `pipeline` للإجابة على الأسئلة باستخدام نموذجك، ومرر النص إليه:
```
Comment (abodacs) [inline docs/source/ar/tasks/question_answering.md:345]:
```suggestion
أبسط طريقة لتجربة نموذجك المُدرَّب للاستدلال هي استخدامه في [`pipeline`]. قم بإنشاء كائن لـ `pipeline` للإجابة على الأسئلة باستخدام نموذجك، ومرِّر النص إليه:
```

File: docs/source/ar/tasks/question_answering.md [hunk 11]
```diff
@@ -0,0 +1,432 @@
...
+```
+
+احصل على أعلى احتمال من مخرجات النموذج لموضعي البداية والنهاية:
+
+```py
+>>> answer_start_index = outputs.start_logits.argmax()
+>>> answer_end_index = outputs.end_logits.argmax()
+```
+
+فك تشفير الرموز المتوقعة للحصول على الإجابة:
```
Comment (abodacs) [inline docs/source/ar/tasks/question_answering.md:390]:
```suggestion
استخلاص الإجابة من الرموز المتوقعة:
```

File: docs/source/ar/tasks/question_answering.md [hunk 12]
```diff
@@ -0,0 +1,432 @@
...
+```
+
+احصل على أعلى احتمال من مخرجات النموذج لموضعي البداية والنهاية:
+
+```py
+>>> answer_start_index = int(tf.math.argmax(outputs.start_logits, axis=-1)[0])
+>>> answer_end_index = int(tf.math.argmax(outputs.end_logits, axis=-1)[0])
+```
+
+فك تشفير الرموز المتوقعة للحصول على الإجابة:
```
Comment (abodacs) [inline docs/source/ar/tasks/question_answering.md:424]:
```suggestion
استخلاص الإجابة من الرموز المتوقعة:
```

**Review by abodacs — COMMENTED** [review]
السﻻم عليكم

تمت المراحعة
@AhmedAlmaghz 
جزاك الله خيرا

**Comment by AhmedAlmaghz** [comment]
أشكرك أخي عبدالله @abodacs  على جهودك العظيمة
تحياتي وتقديري

**Review by stevhliu — APPROVED** [review]
Thanks for your contribution!

**Comment by AhmedAlmaghz** [comment]
Thank you @abodacs  and @stevhliu

**Pull request merged**